core/config/config.go

```diff
@@ -152,9 +152,16 @@ func Load() ConfigurationProvider {
 	}
 	baseCfg := NewProviderGroup("global", static...)
 
+	// TODO: (at) If application ID is not specified, the code lets the service initialization
+	// handle the failures and skips initializing dynamic config.
+	serviceName, ok := baseCfg.GetValue("applicationID").TryAsString()
+	if !ok {
+		return baseCfg
+	}
 	var dynamic []ConfigurationProvider
+
 	for _, providerFunc := range _dynamicProviderFuncs {
-		cp, err := providerFunc(baseCfg)
+		cp, err := providerFunc(serviceName, baseCfg)
```

Comment:
Thanks. This was an abstraction to get serviceName before hand without having to call config.GetValue multiple times. I'll revert this PR and move this fetch downstream.